core/base/document.php

```diff
@@ -1284,4 +1288,29 @@ protected function get_post_statuses() {
 	protected function get_have_a_look_url() {
 		return $this->get_permalink();
 	}
+
+	public function handle_revisions_changed( $post_has_changed, $last_revision, $post ) {
+		// In case default, didn't determine the changes.
+		if ( ! $post_has_changed ) {
+			$last_revision_id = $last_revision->ID;
+			$last_revision_document = Plugin::instance()->documents->get( $last_revision_id );
+			$post_document = Plugin::instance()->documents->get( $post->ID );
+
+			$last_revision_settings = $last_revision_document->get_settings();
+			$post_settings = $post_document->get_settings();
+
+			// TODO: Its better to add crc32 signature for each revision and then only compare one part of the checksum.
+			$post_has_changed = $last_revision_settings !== $post_settings;
+		}
+
+		return $post_has_changed;
+	}
+
+	public function add_handle_revisions_changed() {
+		add_filter( 'wp_save_post_revision_post_has_changed', [ $this, 'handle_revisions_changed' ], 10, 3 );
+	}
+
+	public function remove_handle_revisions_changed() {
+		remove_filter( 'wp_save_post_revision_post_has_changed', [ $this, 'handle_revisions_changed' ] );
+	}
```

Comment:
1. Make them private.
2. Consider naming the methods something like `<add|remove>_handle_revisions_post_has_changed_filter` for better clarifity.

Reply:
Handled.